### PR TITLE
Add Django 2.2, Django 3.0 support & Drop Django 1.5, 1.6, 1.7, 1.8, 1.9, 1.10, 2.0 and 2.1 as well as Python 3.4 and 3.5

### PR DESCRIPTION
**Pull request by joehybird**
This PR is based on the work of @jmerkow (#144) and @sliverc (#142)

### CODE REVIEW

**Comment by Natim** [comment]
Could you remove Python 3.4 and add Python 3.7 and 3.8 ?

File: .travis.yml [hunk 1]
```diff
@@ -1,28 +1,37 @@
 language: python
+dist: xenial
```
Comment (Natim) [inline .travis.yml]: We could go for bionic I guess.

File: .travis.yml [hunk 2]
```diff
@@ -1,28 +1,37 @@
...
-  - TOXENV=flake8
-#  - TOXENV=sphinx
-#  - TOXENV=readme
+  - DJANGO=1.8
+  - DJANGO=1.9
+  - DJANGO=1.10
+  - DJANGO=1.11
+  - DJANGO=2.0
+  - DJANGO=2.1
+  - DJANGO=2.2
```
Comment (Natim) [inline .travis.yml]: We could add DJANGO=3 too.

**Comment by Natim** [comment]
I guess we should drop support for old version of Django and Python 2.7 versions too.

**Comment by Natim** [comment]
image

**Comment by Natim** [comment]
We should drop support for all the red ones here:

image

File: demo/demoproject/compat.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
+from distutils.version import StrictVersion
+from django.utils.version import get_version
+
+
+try:
```
Comment (Natim) [inline demo/demoproject/compat.py]: I guess we could strip down this file to the minimum.

Comment (Natim) [reply demo/demoproject/compat.py]: If we keep support for 1.11 I guess we still need this.

**Comment by Natim** [comment]
Thank you for you work on this, I am glad to see this project being updated.

**Comment by joehybird** [comment]
Thank you @Natim for fixing travis (I'm always stuck by this...). Maybe add python 3.x targets for django 1.11 in tox ?

**Comment by Natim** [comment]
Yes I fixed it in travis.

**Pull request merged**